src/app.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-unused-vars, no-unsafe-finally */
+
+import {Utils} from '@natlibfi/melinda-commons';
+import amqplib from 'amqplib';
+import {logError} from './utils';
+import {EventEmitter} from 'events';
+import {consumeQueue} from './services/fromQueueService';
+
+import {
+	AMQP_URL,
+	NAME_QUEUE_BULK,
+	NAME_QUEUE_PRIORITY,
+	PURGE_QUEUE_ON_LOAD,
+	EMITTER_JOB_CHECK_QUEUE,
+	EMITTER_JOB_CONSUME,
+	NAME_QUEUE_REPLY
+} from './config';
+
+class QueueEmitter extends EventEmitter {}
+export const EMITTER = new QueueEmitter();
+
+const {createLogger} = Utils;
+const logger = createLogger(); // eslint-disable-line no-console
+
+process.on('UnhandledPromiseRejectionWarning', err => {
+	logError(err);
+	process.exit(1);
+});
+
+run();
+
+async function run() {
+	logger.log('info', 'Melinda-rest-import-queue has been started');
+	await operateRabbitQueues(true, PURGE_QUEUE_ON_LOAD, true);
+	setEmitterListeners();
```

Comment:
No need to emit anything. Set up queues when started and then start consuming from queues using channels.